### PR TITLE
Proxy updates

### PR DESCRIPTION
**Pull request by mvid**

### CODE REVIEW

File: cmd/staticProxy.go [hunk 1]
```diff
@@ -92,15 +90,19 @@ func runProxyBlocking() {
...
+	prox := fastproxy.NewFastStaticProxy(fastproxy.FastStaticProxyConfig{
 		MiddleWares: []middleware.GraphqlMiddleware{
 			&middleware.ValidationMiddleware{},
 			&middleware.ContextMiddleware{},
 		},
 		RequestConfigProvider: proxy.NewStaticSchemaProvider(proxy.RequestConfig{
 			Schema:              &schema,
-			BackendAddr:         []byte(staticProxyBackendURL),
-			BackendHost:         staticProxyBackendAddr,
+			BackendURL:          *backendURL,
```
Comment (mvid) [inline cmd/staticProxy.go:105]: Curious as to your opinion on this, but I couldn't find a reason to keep the addr and url separate. Using the stdlib url.URL type, and using the access methods on it as needed.

File: pkg/proxy/proxy.go [hunk 2]
```diff
@@ -2,13 +2,12 @@ package proxy
...
 import (
 	"bytes"
-	"context"
 	"io"
 )
 
 type Proxy interface {
-	AcceptRequest(uri string, body io.ReadCloser, ctx context.Context) (*bytes.Buffer, error)
-	DispatchRequest(input []byte) ([]byte, error)
+	AcceptRequest(contextValues map[string][]byte, requestURI []byte, body io.Reader, buff *bytes.Buffer) error
```
Comment (mvid) [inline pkg/proxy/proxy.go:9]: Updated the interface with the new signature you implemented

File: pkg/proxy/proxy.go [hunk 3]
```diff
@@ -2,13 +2,12 @@ package proxy
...
+	DispatchRequest(buff *bytes.Buffer) (io.ReadCloser, error)
```
Comment (mvid) [inline pkg/proxy/proxy.go:10]: Changed this from returning http.Response, as that would tie this interface to the net/http lib

**Comment by jensneuse** [comment]
Thanks for your contribution.

I've pushed a few minor changes to this branch:
https://github.com/jensneuse/graphql-go-tools/tree/mvid-proxy-update

Please review and comment here if you're okay with it.
If you feel the need for additional commits please PR on branch mvid-proxy-update.

If everything is OK I'll close this PR and merge mvid-proxy-update.

**Comment by mvid** [comment]
Yes, seems good

**Pull request merged**